src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+                var items = GenerateItems(1);
+                list.AddRange(items);
+                Assert.True(list.Contains(items[0]));
+            }
+        }
+
+        [Fact]
+        public void EmptyCollectionAddThenContainsDifferent()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                var list = GetList(null);
+                var items = GenerateItems(1);
+                list.AddRange(items);
+                Assert.False(list.Contains(GenerateItem()));
+            }
+        }
+
+        [Fact]
+        public void EmptyCollectionAddTheContainsNull()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                var list = GetList(null);
+                var items = GenerateItems(1);
+                list.AddRange(items);
+                if (ItemsMustBeNonNull
+                    && SearchingThrowsFromInvalidValue)
+                {
+                    Assert.Throws<ArgumentException>(
+                        () => list.Contains(null));
+                }
+                else
+                {
+                    Assert.False(list.Contains(null));
+                }
+            }
+        }
+
+        [Fact]
+        public void AddAndContainsNullValue()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly && !ItemsMustBeNonNull)
+            {
+                var list = GetList(null);
+                list.Add(null);
+                Assert.True(list.Contains(null));
+            }
+        }
+
+        [Fact]
+        public void ContainsValueThatExistsTwice()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly
+                && !ItemsMustBeNonNull && !ItemsMustBeUnique)
+            {
+                var list = GetList(null);
+                var items = GenerateItems(16);
+                list.Add(items);
+                list.Add(items);
+
+                for (int i = 0; i < items.Length; i++)
+                {
+                    Assert.True(list.Contains(items[i]));
+                }
+                Assert.False(list.Contains(GenerateItem()));
+            }
+        }
+
+        [Fact]
+        public void NullInMiddleContains()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly
+                && !ItemsMustBeNonNull)
+            {
+                var list = GetList(null);
+                var items = new object[33];
+                var tempItems = GenerateItems(32);
+                list.AddRange(tempItems);
+                Array.Copy(tempItems, 0, items, 0, 16);
+                Array.Copy(tempItems, 16, items, 17, 16);
+                list.Insert(16, null);
+                items[16] = null;
+
+                foreach (object item in items)
+                {
+                    Assert.True(list.Contains(item));
+                }
+                Assert.False(list.Contains(GenerateItem()));
+            }
```

Comment:
I'm not sure why the null in the middle is important, but since you don't compare contents of the array and list you don't need to copy the items into a second array.  Just make your N objects, insert a null to the list a N/2, Contains each of the items, and Contains null.

The order of Contains, after all, shouldn't matter :)